### ISSUE TITLE
Inserting directive inline causes current p tag to close

### ISSUE THREAD

**Issue by Valchris** [fraywing/textAngular#906]
I'm attempting to add a toolbar icon for inserting a directive. When the toolbar is clicked and the directive is added, the current line P tag is closed, the directive is added and a new P tag is opened.

Desired output:

    <p> Hello <span special-stuff></span> World</p>

Current output:

    <p> Hello </p>
    <span special-stuff></span>
    <p> World </p>'

Is the desired output possible with the current codebase? Is wrapSelect('insertHTML',...) the correct function for achieving this?

For reference, this is my toolbar icon:

```javascript
app.config(["$provide", function ($provide) {
    $provide.decorator('taOptions', ['$delegate', 'taRegisterTool', '$window', '$compile', function (taOptions, taRegisterTool, $window, $compile) {
        taRegisterTool('assetTag', {
            iconclass: 'fa fa-cube',
            tooltiptext: 'Insert a linked asset',
            action: function () {
                var assetPrompt = $window.prompt('Please enter the asset ID', '');
                if (assetPrompt && assetPrompt !== '') {
                    var embed = '<span asset="' + assetPrompt + '" >' + assetPrompt+ '</span>';
                    $compile(embed)(this);
                    this.$editor().wrapSelection('insertHTML', embed, true);
                }
            }
        });
        return taOptions;
    }]);
}]);
```

**Comment by pbassut** [comment]
Well, the output itself is possible. It's just text anyway, right?
But I assume you want that directive to be compiled.
And here's why it's not gonna happen for now: #51 #52 #62

**Comment by Valchris** [comment]
I've modified the textAngular-sanitize to accept my attribute. I have the directive compiling and working correctly, the issue I'm having is that using the insert HTML causes the paragraph tag to end even if the directive is inserted in the middle of a paragraph. Any suggestions on solving that issue?

Daniel

**Comment by pbassut** [comment]
Could you provide a plunkr so we can investigate it a bit more?

**Comment by Valchris** [comment]
I was able to get it to work, here (http://plnkr.co/edit/t83ch2EGyvWSsPqv5Uhv?p=preview) is a plunkr regardless to share my process. 

I edited validCustomTag in textAngular-sanitize to accept my custom attribute named asset.

It turns out when I had the asset being output as a div it automatically ended the paragraph tag but the same isn't true for a span. I'd love to better understand why/what automatically closes tags within the editor.
